llvm/tools/llvm-exegesis/lib/SerialSnippetGenerator.cpp

```diff
@@ -81,12 +75,10 @@ static ExecutionMode getExecutionModes(const Instruction &Instr,
     EM |= ExecutionMode::ALWAYS_SERIAL_TIED_REGS_ALIAS;
   if (Instr.hasMemoryOperands())
     EM |= ExecutionMode::SERIAL_VIA_MEMORY_INSTR;
-  else {
-    if (Instr.hasAliasingRegisters(ForbiddenRegisters))
-      EM |= ExecutionMode::SERIAL_VIA_EXPLICIT_REGS;
-    if (Instr.hasOneUseOrOneDef())
-      EM |= ExecutionMode::SERIAL_VIA_NON_MEMORY_INSTR;
-  }
+  if (Instr.hasAliasingNotMemoryRegisters(ForbiddenRegisters))
+    EM |= ExecutionMode::SERIAL_VIA_EXPLICIT_REGS;
+  if (Instr.hasOneUseOrOneDef())
+    EM |= ExecutionMode::SERIAL_VIA_NON_MEMORY_INSTR;
```

Comment:
Instructions with memory operands can reach this `if` now due to the removal of the `else` that was around it. Was that intentional? I assume it was intentional for the `hasAliasingNotMemoryRegisters` code.

Reply:
Yes, as example this allows to chain instructions via rd and rs2:
amo* rd, rs2, (rs1)
amo* rd, rs2, (rs1)